webapp/sources/rudder/rudder-core/src/main/scala/com/normation/rudder/services/policies/DeploymentService.scala

```diff
@@ -1077,16 +1082,16 @@ trait PromiseGeneration_BuildNodeContext {
             withDefautls      <- CompareProperties
                                    .updateProperties(
                                      nodeParam.toList.map { case (k, v) => NodeProperty(k, v.value, v.inheritMode, None) },
-                                     Some(nodeInfo.properties)
+                                     Some(nodeInfo.properties.toList)
                                    )
-                                   .map(p => nodeInfo.modify(_.node.properties).setTo(p))
+                                   .map(p => nodeInfo.modify(_.properties).setTo(Chunk.fromIterable(p)))
                                    .toBox
           } yield {
             (
               nodeId,
               InterpolationContext(
-                withDefautls,
-                policyServer,
+                withDefautls.toNodeInfo,
+                policyServer.toNodeInfo,
```

Comment:
`InterpolationContext` and `CompareProperties` are let untouched for now, so we have some conversion happening. It will be clean-up in a further iteration.